kubernetes/resource_kubernetes_stateful_set.go

```diff
@@ -203,3 +227,46 @@ func resourceKubernetesStatefulSetDelete(d *schema.ResourceData, meta interface{
 
 	return nil
 }
+
+// retryUntilStatefulSetRolloutComplete checks if a give job finished its execution and either in Complete or Failed state
+func retryUntilStatefulSetRolloutComplete(conn *kubernetes.Clientset, ns, name string) resource.RetryFunc {
+	return func() *resource.RetryError {
+		res, err := conn.AppsV1().StatefulSets(ns).Get(name, metav1.GetOptions{})
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		// NOTE This is what kubectl uses to determine if a rollout is done
+		// We are using this here because the logic for determining if a StatefulSet
+		// is done is gnarly and we dont want to duplicate it in the provider
```

Comment:
Ha, I like the personality this comment has. Just adding punctuation.

```suggestion
		// is done is gnarly and we don't want to duplicate it in the provider.
```